Update webgazer extract

RobinReborn opened the pull request:
I have updated the code so that webgazerExtract works with facemesh.

I've made a few stylistic changes as well, separating larger files into multiple files, including separating HTMl from CSS and Javascript.

I've made getEyeFeats accessible outside of ridge regression.

I've also changed the names of the .csv files written, I considered deleting the images being written and just keeping the .csvs but I wasn't sure if there was a purpose in storing the images.

Skylion007 commented:
Thanks so much for @RobinReborn , just as an update we are about to merge in #142 later tonight which may cause a lot of merge conflicts in your code so be prepared to rebase and resolve them.

RobinReborn commented:
OK - I've incorporated #142 and #150 into my code.

Skylion007 commented:
Sorry @RobinReborn You are going to need to do one last merge from master. We finally hammered out most of the remaining npm conversion issues. Thank you for your patience!

RobinReborn commented:
OK, I updated the .gitignore but I am still having problems with webgazer.js not being found like I had before (I'm running Chrome on Ubuntu).

The existing conflicts are mainly because I changed ridgeRegWeighted.js to be ridgeRegWeighted.mjs - I also added some things to the .gitignore so I wouldn't upload the data for webgazerExtract

Skylion007 commented:
I am guessing you are on Windows? That's probably the main issue since
symlinks aren't enabled by default. A long term solution is probably to
just install webgazer as a node dependency in www.

On Mon, Aug 10, 2020 at 6:02 PM Robin Gane-McCalla <notifications@github.com>
wrote:

> OK, I updated the .gitignore but I am still having problems with
> webgazer.js not being found like I had before (I'm running Chrome on
> Ubuntu).
>
> The existing conflicts are mainly because I changed ridgeRegWeighted.js to
> be ridgeRegWeighted.mjs - I also added some things to the .gitignore so I
> wouldn't upload the data for webgazerExtract
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/brownhci/WebGazer/pull/149#issuecomment-671612000>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAPVMX7DU65HMPAGTZPK2DDSABU6VANCNFSM4PX53PCQ>
> .
>

RobinReborn commented:
No - I'm on Ubuntu.  I just hadn't renamed webgazer.var.js to webgazer.js it's working now.

Skylion007 commented:
@RobinReborn Is this ready? Should the remaining conflicting files be dealt with?

RobinReborn commented:
It's ready now, I have dealt with the conflicting files.

Skylion007 commented:
So dist2/ folder needs to be removed from this commit.

Skylion007 commented on www/data/src/readme.md:
```diff
@@ -8,7 +8,7 @@ james_tompkin@brown.edu - Any issues, email me.
 After extraction, this makes it simple and efficent to analyse the performance of WebGazer in your favourite _data science_ application.
 
-Works with webgazer.js commit 01a9c68cf0a4b1f3a6d6910bf86d1ff403831ec7 (and onwards) from 30th July 2018 11:49 PM.
+Works with webgazer.js commit 5a12c64db54b489cda9bad5a1b90fa9ccfdde4e8
```
Grr... this commit ID will be changed once it's squashed, this ReadMe change will likely need to be a second PR.

RobinReborn commented:
OK - I've removed dist2/ and my dist/ file structure is the same as what's currently in master.

Skylion007 commented:
Please resolve the deletion conflict. dist/webgazer.commonjs2.js.map still needs to be removed. @RobinReborn Is the new .gitignore not working for you?

Skylion007 commented:
Went ahead and fixed it.

RobinReborn commented:
I merged the .gitignore after everything else, sorry about that.

Do you need any other changes from me?

Skylion007 commented:
@RobinReborn Looks like your refactor did break something. I got this error on P_01
```
webgazerExtractClient.js:359 Uncaught (in promise) TypeError: webgazer.getRegression(...)[0].getEyeFeats is not a function
    at runWebGazerSendResult (webgazerExtractClient.js:359)
```
